### PR TITLE
IndirectDataAnalysis - ConvFit Interface Code Maintenance

### LINKED ISSUE #15105: Indirect Data Analysis ConvFit - Tidy up ConvFit UI code
The ConvFit.cpp Interface code has become rather cluttered. It would be good to spend some time during the maintenance period to tidy this and improve readability.

### PR DESCRIPTION
**Pull request by louisemccann**
Work has been done to generally improve and refactor some of the code for the Indirect ConvFit interface

**To test:**
- Open `ConvFit`(Interfaces > Indirect > Data analysis > ConvFit)
- Sample = `irs26176_graphite002_red.nxs`
- Resolution = `irs26173_graphite002_res.nxs`
- Test different `FitType`

Fixes #15105.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp [hunk 1]
```diff
@@ -1319,14 +1288,14 @@ void ConvFit::singleFitComplete(bool error) {
     pref += "f" + QString::number(subIndex) + ".";
   }
 
-  if (fitTypeIndex == 1 || fitTypeIndex == 2) {
+  if (fitTypeIndex == 2) {
     functionName = "Lorentzian 1";
```
Comment (PranavBahuguna) [inline MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp]: Is there any reason why we can't move the assignment of `functionName` to "Lorentzian 1" to the second if block on line 1295? The first if block doesn't look like its needed anymore.

**Comment by PranavBahuguna** [comment]
Tests seem to run just fine. Code is also mostly good, just see the line comment I added above. Once this is addressed and checks pass, it should be good to go.

**Comment by PranavBahuguna** [comment]
Tests still pass and code looks fine now. All seems good to me :shipit:

File: MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp [hunk 2]
```diff
@@ -1748,44 +1672,88 @@ void ConvFit::updatePlotOptions() {
...
+* @return a QMap populated with name, value pairs for parameters where name =
+* fitFunction.parametername and value is either from the default map or 0
 */
-QString ConvFit::convertBackToShort(const std::string &original) {
-  QString result = QString::fromStdString(original.substr(0, 3));
-  auto pos = original.find(" ");
-  if (pos != std::string::npos) {
-    result += original.at(pos + 1);
+QMap<QString, double>
+ConvFit::constructFullPropertyMap(const QMap<QString, double> defaultMap,
```
Comment (SimonHeybrock) [inline MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp]: Pass by reference?

File: MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp [hunk 3]
```diff
@@ -1748,44 +1672,88 @@ void ConvFit::updatePlotOptions() {
...
+                                  const QStringList parameters,
```
Comment (SimonHeybrock) [inline MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp]: Pass by ref?

File: MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp [hunk 4]
```diff
@@ -462,21 +403,48 @@ void ConvFit::algorithmComplete(bool error) {
...
+ * algorithm runner
+ * @param workspaceName	:: The name of the workspace to add the log to
+ * @param logName		:: The title of the log input
+ * @param logText		:: The information to match the title
+ * @param logType		:: The type of information (String, Number)
+ */
+void ConvFit::addSampleLogsToWorkspace(const std::string workspaceName,
+                                       const std::string logName,
+                                       const std::string logText,
+                                       const std::string logType) {
```
Comment (SimonHeybrock) [inline MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp]: Pass by ref?

**Pull request merged**